### PR TITLE
refactor observables

### PR DESCRIPTION
**Pull request by KATT**
Gonna make our own implementation closer to rxjs..

- Rename `rxjs/*` to `observable/*`
- Export observable-related stuff at `@trpc/client/observable`
- Refactor to export all links straight from `@trpc/client`

### CODE REVIEW

**Comment by KATT** [comment]
I'd like to compare these:

- https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=[{+createTRPCClient+}]
- https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=[{+createTRPCClient+}]

But not able to run bundlejs right now 🤷 .

Seems like there are a lot of `/* PURE */`-annotation things missing where I think they should be, but maybe that doesn't matter either.

**Comment by sachinraja** [comment]
> I'd like to compare these:
https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=[{+createTRPCClient+}]
https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=[{+createTRPCClient+}]
But not able to run bundlejs right now 🤷 .

Talked to okikio and it's back up now (thanks for him for the quick fix). Seems to be a very small increase from alpha.9 to alpha.14 (5.22 to 5.57 KB).

**Comment by KATT** [comment]
> > I'd like to compare these:
> > [https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=[{+createTRPCClient+}]](https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=%5B%7B+createTRPCClient+%7D%5D)
> > [https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=[{+createTRPCClient+}]](https://bundle.js.org/?q=@trpc/client@10.0.0-alpha.14&treeshake=%5B%7B+createTRPCClient+%7D%5D)
> > But not able to run bundlejs right now 🤷 .
> 
> Talked to okikio and it's back up now (thanks for him for the quick fix). Seems to be a very small increase from alpha.9 to alpha.14 (5.22 to 5.57 KB).

Amazing. Seems like tree shaking it works then ☺

File: examples/standalone-server/src/client.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
 import { createTRPCClient } from '@trpc/client';
-import { httpLink } from '@trpc/client/links/httpLink';
-import { splitLink } from '@trpc/client/links/splitLink';
-import { createWSClient, wsLink } from '@trpc/client/links/wsLink';
+import { httpLink } from '@trpc/client';
+import { splitLink } from '@trpc/client';
+import { createWSClient, wsLink } from '@trpc/client';
```
Comment (sachinraja) [inline examples/standalone-server/src/client.ts:5]: duplicate imports

File: packages/client/rollup.config.js [hunk 2]
```diff
@@ -9,8 +9,7 @@ const config = getRollupConfig({
     'src/links/splitLink.ts',
     'src/links/transformerLink.ts',
     'src/links/wsLink.ts',
```
Comment (sachinraja) [inline packages/client/rollup.config.js:11]: These `src/links/*` inputs are no longer needed since they're all being exported from `index.ts` now.

**Review by sachinraja — CHANGES_REQUESTED** [review]
This looks really nice. Left only a couple very minor comments.

**Pull request merged**